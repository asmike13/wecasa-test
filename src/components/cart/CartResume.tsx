import './styles.scss'

export interface ICartResumeProps {
	hasItems: boolean
	totalTime: string
	totalPrice: string
}

const CartResume = ({
	hasItems,
	totalTime,
	totalPrice,
}: ICartResumeProps) => {
	return (
		<>
			{hasItems
				? <>
					<span>Panier :</span>
					<ul>
						<li>{`Prix total : ${totalPrice}`}</li>
						<li>{`Dur√©e totale : ${totalTime}`}</li>
					</ul>
				</>
				: 'Panier vide'
			}
		</>
	)
}

export default CartResume
