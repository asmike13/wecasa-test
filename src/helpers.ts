export const timeFormat = (timeInMinutes: number) => {
	const hours = Math.floor(timeInMinutes / 60)
	const minutes = timeInMinutes % 60

	const displayHours = hours > 0 ? `${hours}H` : ''
	const displayMinutes = minutes > 0 ? `${minutes}min` : ''

	return `${displayHours} ${displayMinutes}`
}

export const priceFormat = (priceInCents: number) => {
	const price = Math.floor(priceInCents / 100)
	const cents = priceInCents % 100

	const displayPrice = price > 0 ? price : ''
	const displayCents = cents > 0 ? `.${cents}` : ''

	return `${displayPrice}${displayCents}â‚¬`
}
