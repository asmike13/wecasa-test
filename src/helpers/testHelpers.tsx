import { Provider } from 'react-redux'
import { store } from 'src/store'

export const StoreWrapper = (props: any) => {
	return (
		<Provider store={store}>
			{props.children}
		</Provider>
	)
}
