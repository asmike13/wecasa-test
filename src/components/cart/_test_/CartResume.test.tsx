import renderer from 'react-test-renderer';
import { StoreWrapper } from 'src/helpers/testHelpers';
import CartResume, { ICartResumeProps } from '../CartResume';

export const props: ICartResumeProps = {
	hasItems: false,
	totalTime: '1H 4min',
	totalPrice: '100â‚¬',
}

it('should render CartResume as snapshot without items', async () => {
	const snapshot = renderer.create(
		<StoreWrapper>
			<CartResume {...props} />
		</StoreWrapper>,
	)

	expect(snapshot).toMatchSnapshot()
})

it('should render CartResume as snapshot with items', async () => {
	const snapshot = renderer.create(
		<StoreWrapper>
			<CartResume {...props} hasItems />
		</StoreWrapper>,
	)

	expect(snapshot).toMatchSnapshot()
})
