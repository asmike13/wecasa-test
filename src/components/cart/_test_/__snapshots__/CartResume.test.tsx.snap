// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CartResume as snapshot with items 1`] = `
Array [
  <span>
    Panier :
  </span>,
  <ul>
    <li>
      Prix total : 100€
    </li>
    <li>
      Durée totale : 1H 4min
    </li>
  </ul>,
]
`;

exports[`should render CartResume as snapshot without items 1`] = `"Panier vide"`;
